File: app/src/main/java/com/durbar/bangabandhuplay/ui/more/MoreActivity.java
package com.durbar.bangabandhuplay.ui.more;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.GridLayoutManager;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;

import com.durbar.bangabandhuplay.MainActivity;
import com.durbar.bangabandhuplay.R;
import com.durbar.bangabandhuplay.databinding.ActivityMoreBinding;
import com.durbar.bangabandhuplay.ui.search.SearchResultActivity;
import com.durbar.bangabandhuplay.utils.Constants;
import com.durbar.bangabandhuplay.utils.NetworkUtil;

public class MoreActivity extends AppCompatActivity {

    private ActivityMoreBinding binding;
    private String id = null, title = null, slug = null;
    private MoreHomeViewModel viewModel;
    private boolean moreSection = false, isHome,isMore = true;
    private NetworkUtil networkUtil;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMoreBinding.inflate(getLayoutInflater());
        viewModel = new ViewModelProvider(this).get(MoreHomeViewModel.class);
        networkUtil = NetworkUtil.getInstance(this);
        setContentView(binding.getRoot());

        id = getIntent().getStringExtra(Constants.CONTENT_ID);
        title = getIntent().getStringExtra(Constants.CONTENT_SECTION_TITLE);
        slug = getIntent().getStringExtra(Constants.CONTENT_SLUG);
        isHome = getIntent().getBooleanExtra(Constants.CONTENT_IS_HOME, false);

        setSupportActionBar(binding.toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeAsUpIndicator(R.drawable.baseline_arrow_back_ios_24);
        getSupportActionBar().setTitle(title);

        if (!networkUtil.isNetworkAvailable()) {
            NetworkUtil.showNoInternetDialog(this);
        }


        if (isHome) {
            if (id != null){
                viewModel.getContentHome(id).observe(this, customContentBySlug -> {
                    try {
                        if (customContentBySlug.getData().getContents() != null) {
                            moreSection = true;
                            binding.watchTrailerRv.setLayoutManager(new GridLayoutManager(this, 2));
                            binding.watchTrailerRv.setAdapter(new MoreHomeAdapter(customContentBySlug.getData().getContents(), this,title,isMore));
                            hideProgressBar();
                        }

                    } catch (Exception e) {
                        e.printStackTrace();
                    }
                });
            }

        } else {
            if (slug != null){
                viewModel.getSingleSubCategoryContents(slug).observe(this, singleSubCategoryRes -> {
                    try {
                        if (singleSubCategoryRes.getData().getOttContents() != null) {
                            moreSection = true;
                            binding.watchTrailerRv.setLayoutManager(new GridLayoutManager(this, 2));
                            binding.watchTrailerRv.setAdapter(new MoreOthersAdapter(singleSubCategoryRes.getData().getOttContents(), this,title,isMore));
                            hideProgressBar();
                        }
                    }catch (Exception e){
                        e.printStackTrace();
                    }
                });
            }

        }


        binding.ivSearch.setOnClickListener(v -> {
            startActivity(new Intent(getApplicationContext(), SearchResultActivity.class));
        });
    }

    private void hideProgressBar() {
        if (moreSection) {
            binding.watchTrailerRv.setVisibility(View.VISIBLE);
            binding.progressBar.setVisibility(View.GONE);
        }
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        startActivity(new Intent(getApplicationContext(), MainActivity.class));
        finish();
        onBackPressed();
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/durbar/bangabandhuplay/ui/more/MoreActivity.kt
package com.durbar.bangabandhuplay.ui.more

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.GridLayoutManager
import com.durbar.bangabandhuplay.MainActivity
import com.durbar.bangabandhuplay.R
import com.durbar.bangabandhuplay.data.model.category.single_sub.SingleSubCategoryRes
import com.durbar.bangabandhuplay.data.model.frontend_custom_content.custom_content_by_slug.CustomContentBySlug
import com.durbar.bangabandhuplay.databinding.ActivityMoreBinding
import com.durbar.bangabandhuplay.ui.search.SearchResultActivity
import com.durbar.bangabandhuplay.utils.Constants

class MoreActivity : AppCompatActivity() {
    private var binding: ActivityMoreBinding? = null
    private var id: String? = null
    private var title: String? = null
    private var slug: String? = null
    private var viewModel: MoreHomeViewModel? = null
    private var moreSection = false
    private var isHome = false
    private val isMore = true
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMoreBinding.inflate(layoutInflater)
        viewModel = ViewModelProvider(this).get(MoreHomeViewModel::class.java)
        setContentView(binding!!.root)
        id = intent.getStringExtra(Constants.CONTENT_ID)
        title = intent.getStringExtra(Constants.CONTENT_SECTION_TITLE)
        slug = intent.getStringExtra(Constants.CONTENT_SLUG)
        isHome = intent.getBooleanExtra(Constants.CONTENT_IS_HOME, false)
        setSupportActionBar(binding!!.toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setHomeAsUpIndicator(R.drawable.baseline_arrow_back_ios_24)
        supportActionBar!!.title = title
        if (isHome) {
            id?.let {
                viewModel!!.getContentHome(it).observe(this) { customContentBySlug ->
                    try {
                        if (customContentBySlug?.data?.contents != null) {
                            moreSection = true
                            binding?.watchTrailerRv?.layoutManager = GridLayoutManager(this, 2)
                            binding?.watchTrailerRv?.adapter =
                                MoreHomeAdapter(
                                    customContentBySlug?.data.contents,
                                    this,
                                    title ?: "",
                                    isMore
                                )
                            hideProgressBar()
                        }
                    } catch (e: Exception) {
                        e.printStackTrace()
                    }
                }
            }
        } else {
            slug?.let {
                viewModel?.getSingleSubCategoryContents(it)?.observe(this) { singleSubCategoryRes ->
                    try {
                        if (singleSubCategoryRes?.data?.ottContents != null) {
                            moreSection = true
                            binding?.watchTrailerRv?.layoutManager = GridLayoutManager(this, 2)
                            binding?.watchTrailerRv?.adapter =
                                MoreOthersAdapter(
                                    singleSubCategoryRes.data.ottContents,
                                    this,
                                    title ?: "",
                                    isMore
                                )
                            hideProgressBar()
                        }
                    } catch (e: Exception) {
                        e.printStackTrace()
                    }
                }
            }
        }
        binding!!.ivSearch.setOnClickListener { v: View? ->
            startActivity(
                Intent(
                    applicationContext, SearchResultActivity::class.java
                )
            )
        }
    }

    private fun hideProgressBar() {
        if (moreSection) {
            binding!!.watchTrailerRv.visibility = View.VISIBLE
            binding!!.progressBar.visibility = View.GONE
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        startActivity(Intent(applicationContext, MainActivity::class.java))
        finish()
        onBackPressed()
        return super.onOptionsItemSelected(item)
    }
}